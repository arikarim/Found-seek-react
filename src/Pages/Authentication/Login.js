import axios from "axios";
import React, { useState } from "react";
import { Redirect, useHistory } from "react-router";
import { alert } from "../../PureFunctions/alert";
import {InvalidAlert} from "../../PureFunctions/InvalidAlert";
import "./registration.css";

const Login = ({ setUser, user, setUser_id }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const history = useHistory();
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const data = await axios.post("http://localhost:3001/users/sign_in", {
        user: {
          email: email,
          password: password,
        },
      });
      setUser_id(data.data.user.id);
      localStorage.setItem("user_id", JSON.stringify(data.data.user.id));
      localStorage.setItem("token", JSON.stringify(data.headers.authorization));
      data.data.message === "You are logged in."
        ? setUser("Logged")
        : setUser("Not Logged");
      history.push("/");
      alert(data);
    } catch (error) {
      InvalidAlert()
      console.log(error);
    }
  };
  if (user === "Logged") {
    return <Redirect to={"/"} />;
  }

  return (
    <div className="login-div">
      <form
        onSubmit={handleSubmit}
        className="my-5 d-flex flex-column  col-10 col-md-8 mx-auto p-5"
      >
        <label htmlFor="email" className="col-form-label my-2">
          Email
        </label>
        <input
          onChange={(e) => setEmail(e.target.value)}
          name="email"
          className="my-3"
          type="email"
          placeholder="email"
          required
        />
        <label htmlFor="password" className="col-form-label my-2">
          Password
        </label>
        <input
          onChange={(e) => setPassword(e.target.value)}
          name="password"
          className="my-3"
          type="password"
          placeholder="password"
          required
        />
        <button className="my-2 btn btndark" type="submit">
          Log in
        </button>
        <a className="text-decoration-none link-dark my-2" href="/signup">
          Sign up
        </a>
      </form>
    </div>
  );
};

export default Login;
