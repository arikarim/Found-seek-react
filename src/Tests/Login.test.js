import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Login from "../Pages/Login";

describe("Header with Link", () => {
  test("should match with snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
