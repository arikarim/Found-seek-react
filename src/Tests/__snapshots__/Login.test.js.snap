// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header with Link should match with snapshot 1`] = `
<div
  className="login-div"
>
  <form
    className="my-5 d-flex flex-column  col-10 col-md-8 mx-auto p-5"
    onSubmit={[Function]}
  >
    <label
      className="col-form-label my-2"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="my-3"
      name="email"
      onChange={[Function]}
      placeholder="email"
      required={true}
      type="email"
    />
    <label
      className="col-form-label my-2"
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="my-3"
      name="password"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
    />
    <button
      className="my-2 btn btndark"
      type="submit"
    >
      Log in
    </button>
    <a
      className="text-decoration-none link-dark my-2"
      href="/signup"
    >
      Sign up
    </a>
  </form>
</div>
`;
